include/common_package/commands.hpp

```diff
@@ -0,0 +1,214 @@
+                        return false;
+
+                    return true;
+                default:
+                    // Every other data type will default to true
+                    break;
+                }
+            }
+
+            // Every other data type will default to true
+            return true;
+        }
+
+        /**
+         * @brief Default constructor for JsonKeyDefinition.
+         */
+        JsonKeyDefinition(){};
+
+        /**
+         * @brief Constructs a JsonKeyDefinition object.
+         *
+         * This constructor initializes a JsonKeyDefinition object with the specified parameters.
+         * Use this constructor if you want to allow multiple data types.
+         *
+         * @param required A boolean indicating whether the key is required or not.
+         * @param data_types A reference to an unordered set of data types.
+         * @param min_val An optional minimum value for the key (default: std::nullopt).
+         * @param max_val An optional maximum value for the key (default: std::nullopt).
+         */
+        JsonKeyDefinition(const bool required, const std::unordered_set<data_type_t> &data_types, std::optional<const float> min_val = std::nullopt, std::optional<const float> max_val = std::nullopt)
+        {
+            this->required = required;
+            this->data_types = data_types;
+            this->min_val = min_val;
+            this->max_val = max_val;
+        }
+
+        /**
+         * @brief Constructs a JsonKeyDefinition object.
+         *
+         * This constructor initializes a JsonKeyDefinition object with the specified parameters.
+         * Use this constructor if you just want to allow one data type.
+         *
+         * @param required A boolean indicating whether the key is required or not.
+         * @param data_type The data type of the key.
+         * @param min_val An optional minimum value for the key (default: std::nullopt).
+         * @param max_val An optional maximum value for the key (default: std::nullopt).
+         */
+        JsonKeyDefinition(const bool required, const data_type_t data_type, std::optional<const float> min_val = std::nullopt, std::optional<const float> max_val = std::nullopt)
+        {
+            this->required = required;
+            this->data_types.insert(data_type);
+            this->min_val = min_val;
+            this->max_val = max_val;
+        }
+    };
+
+    /**
+     * @brief The CommandDefinitions class provides a set of static functions to define and check command keys for different types.
+     *
+     * The CommandDefinitions class contains static functions to define the keys for different commands and check if a JSON string matches the definition.
+     * It provides a map of command keys and their corresponding definitions, as well as a function to check a JSON string against the defined command keys.
+     */
+    class CommandDefinitions
```

Comment:
Würde ich in seperate Header aufteilen.

Reply:
Solange  es nicht zu stark ausartet würde ich es in einem Behalten, da es fest zusammengehört.